src/Sulu/Bundle/SnippetBundle/DependencyInjection/Compiler/SnippetAreaCompilerPass.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SnippetBundle\DependencyInjection\Compiler;
+
+use Sulu\Component\Content\Metadata\StructureMetadata;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Get all snippets from compiler.
+ */
+class SnippetAreaCompilerPass implements CompilerPassInterface
+{
+    /**
+     * @param ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $structureFactory = $container->get('sulu_content.structure.factory');
+        $structures = $structureFactory->getStructures('snippet');
+
+        $locales = $container->getParameter('sulu_core.locales');
+
+        $defaultAreas = [];
+        $areas = [];
+
+        /** @var StructureMetadata $structure */
+        foreach ($structures as $structure) {
+            $template = $structure->getName();
+
+            $templateTitles = [];
+            foreach ($locales as $locale) {
+                $templateTitles[$locale] = $structure->getTitle($locale);
+            }
+
+            $defaultAreas[$template] = [
+                'key' => $template,
+                'template' => $template,
+                'title' => $templateTitles,
+            ];
+
+            foreach ($structure->getAreas() as $area) {
+                $area = $this->getArea($template, $area, $locales, $templateTitles);
+                $areas[$area['key']] = $area;
+            }
+        }
+
+        if (empty($areas)) {
+            $areas = $defaultAreas;
+        }
+
+        $container->setParameter('sulu_snippet.areas', $areas);
+    }
+
+    /**
+     * Get area.
+     *
+     * @return array
+     */
+    private function getArea($template, $area, $locales, $templateTitles)
+    {
+        $key = $template . '.' . $area['key'];
+
+        $titles = [];
+
+        foreach ($locales as $locale) {
+            $title = $templateTitles[$locale] . ' ' . ucfirst($area['key']);
```

Comment:
Isn't it problematic to use the `key` here? That's a value the content manager usually doesn't see...